File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.kt
class Solution {
    fun solution(ingredient: IntArray): Int {
        var answer = 0
        val stack = mutableListOf<Int>()

        for (i in ingredient) {
            stack.add(i)

            if (stack.size >= 4) {
                val len = stack.size
                if (stack[len-4] == 1 && stack[len-3] == 2 &&
                    stack[len-2] == 3 && stack[len-1] == 1) {
                    repeat(4) { stack.removeAt(stack.size - 1) }
                    answer++
                }
            }
        }
        return answer
    }
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    stack = []

    for i in ingredient:
        stack.append(i)

        if len(stack) >= 4:
            if stack[-4:] == [1, 2, 3, 1]:
                # 햄버거 완성 → 마지막 4개 제거
                del stack[-4:]
                answer += 1

    return answer
